src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Runtimes/MissingSdkRuntimeDetector.cs

```diff
@@ -17,6 +17,7 @@ internal class MissingSdkRuntimeDetector : OnceInitializedOnceDisposedAsync, IPr
 
         private Guid _projectGuid;
         private bool _enabled;
+        private static readonly object s_listOfMissingRuntimesLock = new();
```

Comment:
We don't need a new object here to lock on. `s_listOfMissingRuntimes` doesn't change its value, so you can lock on that directly.

Reply:
HashSet was replaced with ConcurrentDictionary

Reply:
HashSet was replaced with ConcurrentDictionary

Reply:
What was the reason for changing it?

Reply:
It is thread-safe

Reply:
So is a locked dictionary. But I see now that the class you've moved this logic too also uses `ConcurrentDictionary<,>` so it makes sense to keep it consistent.